### PR TITLE
[APP-2865] Navigation by app id

### PR DESCRIPTION
**Pull request by pedroagostinhoaptoide**
**What does this PR do?**

   - Changes AppViewScreen navigation to use app id instead of package name when possible.
   - Removes the flag that controls store name usage on app meta requests. The store name is now only appended when the request is done with anything other than app id.
   - Makes the installer notifications use app id instead of package name when creating the deeplink to navigate to the app.

**Database changed?**

   Yes - changed the AppDetails database in order to include the app id. It will be the app id of the version for which an installation was started.

**Where should the reviewer start?**

- [ ] See by commit.

**How should this be manually tested?**

  **Every flow that goes into the app view should be tested. That includes navigating from:**

1. Home Screen
2. See all screens
3. Ahab - atm only notifications
4. APKFY

  Flow on how to test this or QA Tickets related to this use-case: [APP-2865](https://aptoide.atlassian.net/browse/APP-2865)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-2865](https://aptoide.atlassian.net/browse/APP-2865)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[APP-2865]: https://aptoide.atlassian.net/browse/APP-2865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-2865]: https://aptoide.atlassian.net/browse/APP-2865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by GuiFPires — COMMENTED** [review]
Shouldn't the json file be committed as well?

**Review by GuiFPires — APPROVED** [review]

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/ViewModelProvider.kt [hunk 1]
```diff
@@ -58,10 +54,7 @@ fun rememberApp(
 )
 
 @Composable
-fun rememberAppBySource(
-  source: String,
-  useStoreName: Boolean = true,
-): Pair<AppUiState, () -> Unit> = runPreviewable(
+fun rememberAppBySource(source: String): Pair<AppUiState, () -> Unit> = runPreviewable(
```
Comment (G0retZ) [inline feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/ViewModelProvider.kt:57]: Lets make the same migration for `v10` and remove this function, leaving only `rememberApp` that returns by source

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/domain/AppMetaUseCase.kt [hunk 2]
```diff
@@ -2,29 +2,12 @@ package cm.aptoide.pt.feature_apps.domain
...
-
-  suspend fun getMetaInfoBySource(
-    source: String,
-    useStoreName: Boolean = true,
-  ): App =
-    appsRepository.getMetaBySource(
-      source = source,
-      useStoreName = useStoreName && !source.contains("com.appcoins.wallet".toPackageNameParam())
-    )
+  suspend fun getMetaInfoBySource(source: String): App =
```
Comment (G0retZ) [inline feature_apps/src/main/java/cm/aptoide/pt/feature_apps/domain/AppMetaUseCase.kt:11]: Lets call it just `getMetaInfo`

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt [hunk 3]
```diff
@@ -98,63 +97,41 @@ private val tabsList = listOf(
...
-  useStoreName: Boolean = true,
-): String = buildAppViewRouteBySource(appId.toAppIdParam(), useStoreName)
-
-fun buildAppViewRouteBySource(
-  source: String,
-  useStoreName: Boolean = true,
-): String = appViewRoute
-  .replace("{$SOURCE}", source)
-  .replace("{$USE_STORE_NAME}", useStoreName.toString())
+fun buildAppViewRouteByAppId(appId: Long): String = buildAppViewRouteBySource(appId.toAppIdParam())
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt:123]: Lets rename this to simple `buildAppViewRoute`

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt [hunk 4]
```diff
@@ -98,63 +97,41 @@ private val tabsList = listOf(
...
     source = source,
-    useStoreName = useStoreName,
     navigate = navigate,
     navigateBack = navigateBack
   )
 }
 
-fun buildAppViewRoute(packageName: String): String =
-  buildAppViewRouteBySource(packageName.toPackageNameParam())
+fun buildAppViewRouteBySource(source: String): String = appViewRoute.replace("{$SOURCE}", source)
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt:121]: Lets rename this to simple `buildAppViewRoute`

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/ViewModelProvider.kt [hunk 5]
```diff
@@ -182,3 +174,6 @@ fun categoryApps(
 fun String.toPackageNameParam() = "package_name=$this"
 
 fun Long.toAppIdParam() = "app_id=$this"
+
+fun String.appendStoreName(storeName: String) =
```
Comment (G0retZ) [inline feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/ViewModelProvider.kt:178]: Lets have extensions `asSource` for corresponding classes in addition to these on primitives

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/analytics/ApkfyManagerProbe.kt [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+          utmSource = UTM_PROPERTY_APKFY_WITHOUT_UTMS,
+          utmMedium = UTM_PROPERTY_APKFY_WITHOUT_UTMS,
+          utmCampaign = UTM_PROPERTY_APKFY_WITHOUT_UTMS,
+          utmTerm = UTM_PROPERTY_APKFY_WITHOUT_UTMS,
+          utmContent = UTM_PROPERTY_APKFY_WITHOUT_UTMS
+        )
+      }
+    }
+  }
+}
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/analytics/ApkfyManagerProbe.kt:49]: Missing new line

File: feature-mmp/src/main/java/cm/aptoide/pt/feature_mmp/apkfy/di/RepositoryModule.kt [hunk 7]
```diff
@@ -29,8 +29,8 @@ internal object RepositoryModule {
   fun provideApkfyManager(
```
Comment (G0retZ) [inline feature-mmp/src/main/java/cm/aptoide/pt/feature_mmp/apkfy/di/RepositoryModule.kt]: Do we still need this?

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/ViewModelProvider.kt (hunk 5 above)
Comment (G0retZ) [reply feature_apps/src/main/java/cm/aptoide/pt/feature_apps/presentation/ViewModelProvider.kt:150]: Still can we encapsulate this?

File: feature-apkfy/src/main/java/cm/aptoide/pt/feature_apkfy/di/RepositoryModule.kt [hunk 8]
```diff
@@ -46,7 +49,26 @@ internal object RepositoryModule {
...
+  @Singleton
+  fun provideApkfyRepository(@RetrofitMMP retrofitMMP: Retrofit): ApkfyRepository {
+    return AptoideApkfyRepository(
+      mmpRemoteDataSource = retrofitMMP.create(AptoideApkfyRepository.Retrofit::class.java)
+    )
+  }
+
+  @Provides
+  @Singleton
+  fun provideApkfyManager(
```
Comment (G0retZ) [inline feature-apkfy/src/main/java/cm/aptoide/pt/feature_apkfy/di/RepositoryModule.kt:64]: Is this necessary?

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/domain/AppMetaUseCase.kt [hunk 9]
```diff
@@ -2,29 +2,14 @@ package cm.aptoide.pt.feature_apps.domain
...
 
-  suspend fun getMetaInfoByAppId(
-    appId: Long,
-    useStoreName: Boolean = true,
-  ): App =
-    appsRepository.getMetaBySource(
-      source = appId.toAppIdParam(),
-      useStoreName = useStoreName
-    )
+  suspend fun getMetaInfo(source: String): App =
```
Comment (G0retZ) [inline feature_apps/src/main/java/cm/aptoide/pt/feature_apps/domain/AppMetaUseCase.kt:11]: Why do we need both of these functions? 🤔

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/domain/AppMetaUseCase.kt [hunk 10]
```diff
@@ -2,29 +2,14 @@ package cm.aptoide.pt.feature_apps.domain
...
 
-  suspend fun getMetaInfoBySource(
-    source: String,
-    useStoreName: Boolean = true,
-  ): App =
-    appsRepository.getMetaBySource(
-      source = source,
-      useStoreName = useStoreName && !source.contains("com.appcoins.wallet".toPackageNameParam())
-    )
+  suspend fun getMetaInfo(appSource: AppSource): App = getMetaInfo(source = appSource.asSource())
```
Comment (G0retZ) [inline feature_apps/src/main/java/cm/aptoide/pt/feature_apps/domain/AppMetaUseCase.kt:14]: Why do we need both of these functions? 🤔

**Review by G0retZ — CHANGES_REQUESTED** [review]

**Pull request merged**